### PR TITLE
[CP-1610] Invalid message when restore backup password is invalid

### PR DESCRIPTION
**Pull request by wojcik92michal**
Jira: [CP-1610]

**Description**

Scope:
* add "invalid password" modal for restore backup flow (see attached image)
* improve "ResultObject" - now it has safer types for failed result

<details>
<summary><b>Screenshots</b></summary>

image

</details>

[CP-1610]: https://appnroll.atlassian.net/browse/CP-1610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lkowalczyk87 — DISMISSED** [review]

File: packages/app/src/index-storage/observers/index-storage-loading.observer.test.ts [hunk 1]
```diff
@@ -36,12 +37,18 @@ describe("Method: observe", () => {
...
 
   test("calls the `DeviceService.request` and `IndexStorageService.loadIndex` methods when `DeviceServiceEvent.DeviceUnlocked` has been emitted", async () => {
-    deviceManager.device.request = jest.fn().mockResolvedValueOnce({
-      data: {
+    // deviceManager.device.request = jest.fn().mockResolvedValueOnce({
+    //   data: {
+    //     deviceToken: "1234567890",
+    //     serialNumber: "0000000000",
+    //   },
+    // })
```
Comment (dkarski) [inline packages/app/src/index-storage/observers/index-storage-loading.observer.test.ts:45]: to remove? 🤔

File: packages/app/src/messages/services/message.service.ts [hunk 2]
```diff
@@ -149,6 +149,7 @@ export class MessageService {
 
     const isAnyErrorResponseFound = successResponses.length !== messages.length
 
+    // TODO [mw] kurwa gowno!
```
Comment (dkarski) [inline packages/app/src/messages/services/message.service.ts:152]: image

**Review by dkarski — COMMENTED** [review]
Changes are fine ✅ However, I can't just let go of one comment 😉

File: packages/app/src/messages/services/message.service.ts (hunk 2 above)
Comment (wojcik92michal) [reply packages/app/src/messages/services/message.service.ts:152]: 😅

File: packages/app/src/index-storage/observers/index-storage-loading.observer.test.ts (hunk 1 above)
Comment (wojcik92michal) [reply packages/app/src/index-storage/observers/index-storage-loading.observer.test.ts]: Removed in https://github.com/mudita/mudita-center/pull/1078/commits/5d1aa7079b4c1c5da6ad60388501daeb034f7b20

File: packages/app/src/messages/services/message.service.ts (hunk 2 above)
Comment (wojcik92michal) [reply packages/app/src/messages/services/message.service.ts]: Removed in https://github.com/mudita/mudita-center/pull/1078/commits/5d1aa7079b4c1c5da6ad60388501daeb034f7b20

**Review by dkarski — APPROVED** [review]

**Comment by lkowalczyk87** [comment]
I cannot create backup (tested on 1.4.0 and 1.5.0 MOS), and icon color is incorrect.

**Review by lkowalczyk87 — APPROVED** [review]

**Comment by wojcik92michal** [comment]
> I cannot create backup (tested on 1.4.0 and 1.5.0 MOS), and icon color is incorrect.

@lkowalczyk87 
1. For this moment, creating backups works for 1.6.0 or above.
2. Regarding color - good catch :) Indeed in the task the color is different, but this is the screenshot from the previous version, so the current color is fine.

**Pull request merged**